stdlib/Printf/test/runtests.jl

```diff
@@ -2,64 +2,307 @@
 
 using Test, Printf
 
-# this macro tests for exceptions thrown at macro expansion
-macro test_me(ty, ex)
-    return quote
-        @test_throws $(esc(ty)) try
-            $(esc(ex))
-        catch err
-            @test err isa LoadError
-            @test err.file === $(string(__source__.file))
-            @test err.line === $(__source__.line)
-            rethrow(err.error)
-        end
+@testset "dynamic" begin
+
+    # dynamic width and precision
+    @test (Printf.@printf("%*d", 10, 12)) == Printf.@printf("%10d", 12)
+    @test (Printf.@printf("%.*d",  4, 12)) == Printf.@printf("%.4d", 12)
+    @test (Printf.@printf("%*.*d", 10, 4, 12)) == Printf.@printf("%10.4d", 12)
+    @test (Printf.@printf("%+*.*d", 10, 4, 12)) == Printf.@printf("%+10.4d", 12)
+    @test (Printf.@printf("%0*.*d", 10, 4, 12)) == Printf.@printf("%010.4d", 12)
```

Comment:
Better, I think, to write these tests very concretely.  For example,

```julia
@test (Printf.@printf("%*d", 10, 12)) == "        12"
```

This way it's super clear what the expected result is because it's right there in the test case.

Reply:
Note also that `@printf` always returns `nothing` and prints to stdout as a side effect.

So these tests will always pass!  (you need `@sprintf`)

Reply:
stupid me :-)